File: samples/Elixir/99-bottles-of-beer.ex
defmodule Bottles do
  def run do
    Enum.each 99..1, fn idx ->
      IO.puts "#{idx} bottle#{plural(idx)} of beer on the wall"
      IO.puts "#{idx} bottle#{plural(idx)} of beer"
      IO.puts "Take one down, pass it around"
      IO.puts "#{idx - 1} bottle#{plural(idx-1)} of beer on the wall"
      IO.puts ""
    end
  end

  def plural(1), do: ""
  def plural(num), do: "s"
end

Bottles.run

File: samples/Go/99-bottles-of-beer.go
package main

import "fmt"

func main() {
	cardinality := func (i int) string {
		if i!=1 {
			return "s"
		}
		return ""
	}
	for i := 99; i > 0; i-- {
		fmt.Printf("%d bottle%s of beer on the wall\n", i, cardinality(i))
		fmt.Printf("%d bottle%s of beer\n", i, cardinality(i))
		fmt.Printf("Take one down, pass it around\n")
		fmt.Printf("%d bottle%s of beer on the wall\n", i-1, cardinality(i-1))
	}
}
